### PR TITLE
More traits, optional serialization, and NotNaN changes

### LINKED ISSUE #7: Hash
`#[derive(Hash)]` is used for the wrappers here, but f32 and f64 don't impl Hash because of NaN (or maybe other reasons)? Seems like the wrappers could manually provide unconditional Hash impls.

### PR DESCRIPTION
**Pull request by AerialX**
Not sure how you'll feel about these changes, I've found them to be pretty useful though!

Closes #7

### CODE REVIEW

**Comment by reem** [comment]
Not sure why I dropped this.. Feel free to rebase/update for latest serde/rustc-serialize/rust and I would love to merge.

**Comment by AerialX** [comment]
@reem rebased!

File: src/lib.rs [hunk 1]
```diff
@@ -55,54 +76,235 @@ impl<T: Float + Copy + PartialEq> PartialEq for OrderedFloat<T> {
...
+}
+
+impl Into<f64> for NotNaN<f64> {
+    fn into(self) -> f64 {
+        self.into_inner()
+    }
+}
+
+impl<T: Float> From<T> for NotNaN<T> {
+    fn from(v: T) -> Self {
```
Comment (reem) [inline src/lib.rs]: Can you add a doc-comment here that this implementation panics if v is NaN?

**Comment by reem** [comment]
Looks great, just add that last comment and this is good to go.

**Comment by AerialX** [comment]
How's that look?

**Comment by reem** [comment]
Woops, gave this another merge conflict, sorry. Looks good though.

**Comment by AerialX** [comment]
Rebased again!

**Comment by reem** [comment]
Thanks!

**Pull request merged**